internal/blueprint/internal/scheme/transform.go

```diff
@@ -0,0 +1,315 @@
+		},
+	}
+
+	if has(obj, "services") {
+		for _, s := range getSlice(obj, "services") {
+			files = append(files, path.Join(s.(string), "service.yaml"))
+		}
+	} else {
+		files = append(files, "services/*/service.yaml")
+	}
+
+	if has(obj, "environments") {
+		for _, e := range getSlice(obj, "environments") {
+			files = append(files, path.Join(e.(string), "environment.yaml"))
+		}
+	} else {
+		files = append(files, "environments/*/environment.yaml")
+	}
+
+	return map[string]any{
+		"apiVersion": "g2a-cli/v2.0",
+		"kind":       "Project",
+		"name":       or(get(obj, "name"), "project"),
+		"files":      files,
+		"tasks":      map[string]any{},
+		"variables":  map[string]any{},
+	}
+}
+
+func toV2Service(obj any) any {
+	artifacts := getSlice(obj, "build", "artifacts")
+
+	if hooks := get(obj, "hooks", "pre-build"); hooks != nil {
+		artifacts = prepend(artifacts, map[string]any{
+			"script": hooksToScript(hooks),
+			"push":   false,
+		})
+	}
+	if hooks := get(obj, "hooks", "post-build"); hooks != nil {
+		artifacts = append(artifacts, map[string]any{
+			"script": hooksToScript(hooks),
+			"push":   false,
+		})
+	}
+
+	releases := getSlice(obj, "deploy", "releases")
+
+	if hooks := get(obj, "hooks", "pre-deploy"); hooks != nil {
+		releases = prepend(releases, map[string]any{
+			"script": hooksToScript(hooks),
+		})
+	}
+	if hooks := get(obj, "hooks", "post-deploy"); hooks != nil {
+		releases = append(releases, map[string]any{
+			"script": hooksToScript(hooks),
+		})
+	}
+
+	tags := []any{}
+
+	for k, v := range getMap(obj, "build", "tagPolicy") {
+		tags = append(tags, map[string]any{k: v})
+	}
+
+	return map[string]any{
+		"apiVersion": "g2a-cli/v2.0",
+		"kind":       "Service",
+		"name":       getString(obj, "name"),
+		"tags":       tags,
+		"artifacts":  artifacts,
+		"releases":   releases,
+		"tasks":      map[string]any{},
+	}
+}
+
+func toV2Environment(obj any) any {
+	return map[string]any{
+		"apiVersion":     "g2a-cli/v2.0",
+		"kind":           "Environment",
+		"name":           getString(obj, "name"),
+		"deployServices": getSlice(obj, "deployServices"),
+		"variables":      getMap(obj, "variables"),
+	}
+}
+
+func hooksToScript(hooks any) any {
+	sh := "set -e"
+	for _, hook := range getSlice(hooks) {
+		sh += fmt.Sprintf("\n%s", hook)
+	}
+	return map[string]any{"sh": sh}
+}
+
+func getString(v any, path ...any) string {
+	str, err := dyno.GetString(v, path...)
+	if err != nil {
+		return ""
+	}
+	return str
+}
+
+func isString(v any, path ...any) bool {
+	_, err := dyno.GetString(v, path...)
+	return err == nil
+}
+
+func getSlice(v any, path ...any) []any {
+	res, err := dyno.GetSlice(v, path...)
+	if err != nil {
+		return []any{}
+	}
+	return res
+}
+
+func getMap(v any, path ...any) map[string]any {
+	res, err := dyno.GetMapS(v, path...)
+	if err != nil {
+		return map[string]any{}
+	}
+	return res
+}
+
+func get(v any, path ...any) any {
+	val, err := dyno.Get(v, path...)
+	if err != nil {
+		return nil
```

Comment:
why do this? I understand generic structures but this simply feels wrong. You just silence everything.